### PR TITLE
UTY-1174: Added missing commands parameters (context, timeout) to MonoBehavior API

### PR DESCRIPTION
**Pull request by MPeti01**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Was even more straightforward than I expected, not sure why we didn't have this already.
#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?
#### Documentation
Added note to WIP GDoc, will start writing it now
#### Primary reviewers
@JonasImprobable @jamiebrynes7

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt [hunk 1]
```diff
@@ -54,8 +54,8 @@ namespace <#= qualifiedNamespace #>
                 }
             }
         }
-<# } #>
 
+<# } #>
```
Comment (JonasImprobable) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: We' still need a blank line between here and `public partial class Requirable` though, right ?

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt [hunk 2]
```diff
@@ -83,15 +83,18 @@ namespace <#= qualifiedNamespace #>
...
 
 <# foreach(var commandDetails in commandDetailsList) { #>
-                public void Send<#= commandDetails.CommandName #>Request(EntityId entityId, <#= commandDetails.FqnRequestType #> request)
+                public long Send<#= commandDetails.CommandName #>Request(EntityId entityId, <#= commandDetails.FqnRequestType #> payload,
+                    uint? timeoutMillis = null, bool allowShortCircuiting = false, object context = null)
                 {
                     if (!VerifyNotDisposed())
                     {
-                        return;
+                        return -1;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: Need to make sure this is well documented

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt (hunk 1 above)
Comment (MPeti01) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: Look at the generated code, it works. (The blank line is 57)

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt (hunk 2 above)
Comment (MPeti01) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: Made note on GDoc

**Pull request merged**